packages/check-time/extra/check-time.c

```diff
@@ -0,0 +1,41 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/timex.h>
+#include <string.h>
+#include <errno.h>
+
+#define USEC_PER_MSEC 1000L
+#define MAX_EST_ERROR_US 100L * USEC_PER_MSEC /* 100 millisecond */
+
+int main() {
+	struct timex tx;
+	int rc;
+	long long int error;
+
+	if (getenv("IGNORE_CHECK_TIME")) {
```

Comment:
practically we should really hard error if IGNORE_CHECK_TIME isn't set. we can make a strong guarantee, and if it isn't it's an internal consistency bug (IGN0RE_CHECK_TIME)